### DISCUSSION TITLE
Update Search filters by PowerShell on several pages

### DISCUSSION THREAD

**Discussion by HeKr09** [microsoft-search/pnp-modern-search#3024] [Ideas]
**Version used**
4.1

**Question**
I have got 60 pages with a search box, search results and search filters. The filters all have the same configuration. The customer now wants to have 5 new filters in all 60 pages. Is there a chance to run a script to configure all 60 web parts at once? I don't want to configure it manually. I tried a PowerShell script but it was not successful.

**Additional context**
I am using the following PowerShell.
```
Connect-PnPOnline -Url https://customer.sharepoint.com/sites/site -Interactive
$sourcePage = Get-PnPPage -Identity scource.aspx
$targetPage = Get-PnPPage -Identity target.aspx
Set-PnPClientSideWebPart -Page $targetPage -Identity 2e61d94b-c269-4fd0-8586-22f6f9865563 -PropertiesJson $sourcePage.Controls[5].PropertiesJson
```
Although the script runs no filters are shown in the target page. The web part is changed and the configuration is stored as I can see in edit mode but there are no filters in read mode.
Anyone an idea?

**Comment by wobba** [comment]
Should be doable. Try a manual edit and compare with the JSON you set, and remember to republish the page for end-users.

**Comment by FranckyC** [comment]
Hi @HeKr09, did you success with your PowerShell script? Can we close this issue? Thanks

**Comment by HeKr09** [comment]
Sorry I did not succeed right now. Could you let it open please?

**Comment by wobba** [comment]
I'll tag it as documentation if someone wants to contribute a sample on how to do this.

**Comment by kasperbolarsen** [comment]
Hi @HeKr09 , so you want to update the configuration of the PnP Search Refiner on a specific set of pages based on the configuration on your "master/template" page? Is that correct?
